_drafts/practical_zephyr_02_dt_basics.md

```diff
@@ -0,0 +1,1468 @@
+
+<!-- excerpt end -->
+
+> **Disclaimer:** This article covers devicetree in _detail_ and from first principles. If you're already somewhat familiar with devicetree or if you don't want to know how it works in detail, this article may not be for you. Instead, consider tuning in for the next article!
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## Prerequisites
+
+This article is part of an article _series_. In case you haven't read the previous articles, please go ahead and have a look. The [first article]({% post_url 2024-01-10-practical_zephyr_basics %}) explains the required installation steps, and we'll follow along with our running UART example from the [previous article]({% post_url 2024-01-24-practical_zephyr_kconfig %}). If you're already familiar with Zephyr and have a working installation at hand, it should be easy enough for you to follow along with your own setup and without reading the previous articles.
+
+> **Note:** A full example application including all the devicetree source files that we'll see throughout this article is available in the [`02_devicetree_basics` folder of the accompanying GitHub repository](https://github.com/lmapii/practical-zephyr/tree/main/02_devicetree_basics).
+
+We'll again be using the [development kit for the nRF52840](https://www.nordicsemi.com/Products/Development-hardware/nrf52840-dk) as a reference and explore its files in Zephyr, but you can follow along with any target - even virtual ones.
+
+A short heads-up: We will _not_ define our own, custom _board_, and we will _not_ describe memory layouts. Those are more advanced topics and go beyond this series. With this article, we just want to have a detailed look and familiarize ourselves with devicetree and the syntax of its input files.
+
+
+## What's a _devicetree_?
+
+Let's first deal with the terminology: In simple words, the _devicetree_ is a tree data structure that you provide to describe your hardware. Each _node_ describes one _device_, e.g., the UART peripheral that we used for logging via `printk` in the [previous article]({% post_url 2024-01-24-practical_zephyr_kconfig %}). Except for the root note, each node has exactly one parent, thus the term device*tree*.
+
+Devicetree files use their own _DeviceTree Source (DTS) format_, defined in the [official devicetree specification](https://www.devicetree.org/specifications/). For certain file types (bindings) used by _devicetree_, Zephyr uses yet another file format - but fear not, it simply replaces the _DTS format_ with simple `.yaml` files. We'll cover _bindings_ in the next article, so for now we'll just focus on devicetree source files. There are also some subtle differences between the official devicetree specification and the way it is used in Zephyr, but we'll touch up on that throughout this article.
```

Comment:
I think here I kept the uppercase T due to the abbreviation. But yeah, it's a bit of a mess. There's all of _device-tree_, _device tree_, _devicetree_, _DeviceTree_ and _Devicetree_ out there :D 

My favourite would be _Devicetree_ since then it is kind of in line with the _CMake_ and _Kconfig_ references that I used in the past articles. Maybe I'll give it another pass!